frontend/src/app/shared/components/file-links/file-link-list/file-link-list-item.component.ts

```diff
@@ -106,4 +111,24 @@ export class FileLinkListItemComponent implements OnInit, AfterViewInit {
       );
     }
   }
+
+  private derivePermissionState():void {
+    switch (this.fileLink._links.permission.href) {
+      case 'urn:openproject-org:api:v3:file-links:permission:View':
+        this.showFloatingActions = true;
+        this.text.floatingText = '';
+        break;
+      case 'urn:openproject-org:api:v3:file-links:permission:NotAllowed':
+        this.showFloatingActions = false;
+        this.text.floatingText = this.i18n.t('js.label_file_link_no_permission');
```

Comment:
Instead of adding the text strings here, please add them at the top (where the text is defined). Here you can that write something like `this.floatingText = this.text.foo`. You'd then need a public variable for that. However, it is more clear to have all used I18n strings in one place and not distributed all over the code.

Reply:
fixed